packages/core/crates/core-types/src/channels.rs

```diff
@@ -125,31 +125,23 @@ impl ChannelEntry {
     }
 
     /// Checks if the closure time of this channel has passed.
-    pub fn closure_time_passed(&self, current_timestamp_ms: u64) -> Option<bool> {
-        assert!(current_timestamp_ms > 0, "invalid timestamp");
-        // round clock ms to seconds
-        let now_seconds: U256 = U256::from(current_timestamp_ms) / 1000u64.into();
-
-        if self.closure_time.eq(&U256::zero()) {
-            None
-        } else {
-            Some(self.closure_time < now_seconds)
-        }
+    /// Also returns `false` if the channel closure has not been initiated.
+    pub fn closure_time_passed(&self, current_timestamp_ms: u64) -> bool {
+        self.remaining_closure_time(current_timestamp_ms)
+            .map(|remaining| remaining == 0)
+            .unwrap_or(false)
     }
 
     /// Calculates the remaining channel closure grace period in seconds.
+    /// Returns `None` if the channel closure has not been initiated yet.
     pub fn remaining_closure_time(&self, current_timestamp_ms: u64) -> Option<u64> {
         assert!(current_timestamp_ms > 0, "invalid timestamp");
         // round clock ms to seconds
-        let now_seconds = U256::from(current_timestamp_ms) / 1000u64.into();
+        let now_seconds = current_timestamp_ms / 1000_u64;
```

Comment:
Maybe worth looking at something like this: https://docs.rs/metrics/latest/metrics/enum.Unit.html

Reply:
https://crates.io/crates/simple-si-units